Objective-C/Internal/CBLLock.h

```diff
@@ -0,0 +1,21 @@
+//
+//  CBLLock.h
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 11/22/17.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+
+#ifndef CBLLock_h
+#define CBLLock_h
+
+#if CBL_THREADSAFE
```

Comment:
If `CBL_THREADSAFE` is undefined or 0, the code won't compile because `CBL_LOCK` is undefined. So is there any point in having `CBL_THREADSAFE` at all, since it needs to be `1`?

Reply:
Just in case we want to turn it off to measure the performance.